crates/holochain/tests/tests/countersigning.rs

```diff
@@ -297,6 +299,383 @@ async fn retry_countersigning_commit_on_missing_deps() {
     wait_for_completion(bob_rx, preflight_request.app_entry_hash).await;
 }
 
+// Checks that when there are multiple authorities returning signature bundles and multiple sessions
+// happening between a pair of agents, the extra signature bundles beyond the first one received
+// are correctly handled. I.e. receiving further signature bundles after a new session has started
+// will not impact the new session.
+// This is trying to check a race condition, so it's probably expected that the test won't always
+// correctly test the problem. However, it is a scenario we've seen fail in Wind Tunnel, so it's
+// worth trying to test here. Seeing flakes from this test is likely a signal that something has
+// regressed.
```

Comment:
Is it worth trying to run this as multiple parallel tasks all doing the same thing to be more likely to produce the conditions without adding too much additional test time?

Reply:
This test was originally added on #4148 and it is broken before the locking change. So I'm happy that it prevents the regression that it's supposed to. I also agree that we should test with multiple agents doing the same thing at the same time but that's pretty much what the Wind Tunnel test does and it's a right pain to write a test that does it correctly so I'm tempted to say that we leave checking the more realistic behavior to Wind Tunnel. Sound okay?

Reply:
Yep, for sure that makes sense :+1: